tasks/gitHelpers.ts

```diff
@@ -0,0 +1,193 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Octokit } from '@octokit/rest';
+import { NugetPackageInfo, platformSpecificPackages } from './offlinePackagingTasks';
+import { PlatformInformation } from '../src/shared/platform';
+
+export interface GitOptions {
+    commitSha: string;
+    targetRemoteRepo: string;
+    baseBranch: string;
+}
+
+export interface BranchAndPROptions extends GitOptions {
+    githubPAT: string;
+    dryRun: boolean;
+    newBranchName: string;
+    userName?: string;
+    email?: string;
+}
+
+export function logError(message: string): void {
+    console.log(`##vso[task.logissue type=error]${message}`);
+}
+
+export async function git(args: string[], printCommand = true): Promise<string> {
```

Comment:
Some of this is now similar to https://github.com/dotnet/vscode-csharp/blob/main/tasks/gitTasks.ts. Can we please move unique functions from here to there please? `getCommitFromNugetAsync` seems like the main thing that is missing. Although, I don't think that is a function of git in any way, so maybe that needs another home.